client/src/components/SignInModal.jsx

```diff
@@ -0,0 +1,75 @@
+import { useContext, useState } from "react"
+import AppContext from "./AppContext"
+import Modal from "react-bootstrap/Modal"
+import Button from "react-bootstrap/Button"
+import Alert from 'react-bootstrap/Alert'
+
+/**
+ * Created the modal component and controls the request to sign a user in.
+ * @param {boolean, function} showSignIn, whether to be open or not, setShowSignIn to close the modal.
+ * @returns the modal component for signing in.
+ */
+export default function SignInModal({ showSignIn, setShowSignIn }) {
+  const { handleSignIn } = useContext(AppContext)
+  const [alertShow, setAlertShow] = useState(false)
```

Comment:
I'd probably name this `alertShowing` so it's not a verb, but is state. Or perhaps `alertIsShowing` or `alertIsVisible`